js/plugins/dotprompt/src/prompt.ts

```diff
@@ -197,16 +198,41 @@ export class Dotprompt<Variables = unknown> implements PromptMetadata {
     return this._generateOptions<O>(opt);
   }
 
+  async renderInNewSpan<O extends z.ZodTypeAny = z.ZodTypeAny>(
+    opt: PromptGenerateOptions<Variables>
+  ): Promise<GenerateOptions<z.ZodTypeAny, O>> {
+    return runInNewSpan(
+      {
+        metadata: {
+          name: this.name,
```

Comment:
I think we need like a hash of the template or something here, to serve as a proxy for version

Reply:
Done